### DISCUSSION TITLE
Abbreviation of file name extensions

### DISCUSSION THREAD

**Discussion by AndreasMatthias** [latex3/l3build#404] [Q&A]
The documentation explains the meaning of the letters `t` and `e` in `.lvt` and `.lve` respectively. But what's the meaning of the other letters? What do the abbreviations `.tlg`, `.pvt`, and `tpf` mean?

I'm having a really hard time remembering these seemingly arbitrary file name extensions.

**Comment by josephwright** [comment]
`.tlg` is 'test log', `pvt` matches `lvt` but is PDF-based ('PDF verification test'), `tpf` matches `tlg` but is PDF based ('test PDF')

**Reply by josephwright** [reply]
(We've used `.tlg` for ever: I modelled `pvt` and `.tpf` on the `.lvt`/`.tlg` pair)

**Reply by AndreasMatthias** [reply]
Oh, I see. `v` for verification was a hard one. Thank you.

Is an `.lve` file a test file like `.lvt` or an expected result file like `.ltg`?

The source code mentions `.lte` as well. Is this a typo? See https://github.com/latex3/l3build/blob/main/l3build-manifest-setup.lua#L157

(I'm just beginning to catch up on l3build.)

**Reply by josephwright** [reply]
`.lve` is a 'LaTeX verification expectation' file - it's an alternative to saving a `.tlg`. You can write an `.lve` that produces the same output as you want from your `.lvt`, then `l3build` can compare the results. The idea is for say a higher-level implementation tested in a `.lvt` against a lower-level equivalent in an `.lve`. The team don't really use this mechanism, but it was requested so we added it.

**Comment by AndreasMatthias** [comment]
Thanks.

**Comment by cfr42** [comment]
@josephwright  that looks like a typo in the comment.

I prefer `v` for 'validity'. Not a fan of A. J. Ayer.
